### PR TITLE
Logging for Reindex and SP Dashboard Data

### PR DESCRIPTION
**Pull request by PTaladay**
## Description
Added logging into the update handler for search parameters to log the status changes and error results for update requests.
Updated Reindex to log the current progress for each reindex job.
These will be used to create/updates dashboards in the future.

## Related issues
Addresses [issue #AB101610].

## Testing
No testing.

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs [hunk 1]
```diff
@@ -779,13 +781,13 @@ private async Task UpdateParametersAndCompleteJob()
                 }
             }
 
-            _logger.LogTrace("Completing reindex job. Updating the status of the fully indexed search parameters: '{ParamUris}'", string.Join("', '", fullyIndexedParamUris));
+            _logger.LogTrace("Reindex job updating the status of the fully indexed search, id: {Id}, parameters: '{ParamUris}'", _reindexJobRecord.Id, string.Join("', '", fullyIndexedParamUris));
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs:784]: Nit: Should this be a LogInformation to be consistant?

**Review by LTA-Thinking — DISMISSED** [review]

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs (hunk 1 above)
Comment (PTaladay) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs:784]: Yeah, I can update it to that.

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs [hunk 2]
```diff
@@ -576,6 +577,7 @@ private async Task CheckJobCompletionStatus()
                 q.Status == OperationStatus.Queued ||
                 q.Status == OperationStatus.Running).Any())
             {
+                _logger.LogInformation("Reindex job stauts update, id: {Id}. Status: {Status}. Progress: {Progress}", _reindexJobRecord.Id, _reindexJobRecord.Status, _reindexJobRecord.Progress);
```
Comment (v-mserdar) [inline src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs:580]: typo - "stauts" should be "status"

Comment (PTaladay) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs]: Thanks. Updated.

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**